packages/containerd/3001-cri-set-a-default-RLIMIT_NOFILE-of-1048576.patch

```diff
@@ -0,0 +1,83 @@
+From 2a4e87269d502ec007ef6505222f34b0bd3774c1 Mon Sep 17 00:00:00 2001
+From: Zac Mrowicki <mrowicki@amazon.com>
+Date: Thu, 22 Oct 2020 20:44:38 +0000
+Subject: [PATCH] cri: set a default RLIMIT_NOFILE of 1048576
+
+The `cri` plugin currently inherits the limit from the default OCI spec
+or the containerd process.  This change sets the default RLIMIT_NOFILE
+to 1048576 in the OCI spec for any container spawned using `cri`.
+---
+ vendor/github.com/containerd/cri/pkg/config/config.go |  4 ++++
+ .../containerd/cri/pkg/containerd/opts/spec.go        | 11 +++++++++++
+ .../containerd/cri/pkg/server/container_create.go     | 11 +++++++++++
+ 3 files changed, 26 insertions(+)
+
+diff --git a/vendor/github.com/containerd/cri/pkg/config/config.go b/vendor/github.com/containerd/cri/pkg/config/config.go
+index 7c5f9eb..4374a62 100644
+--- a/vendor/github.com/containerd/cri/pkg/config/config.go
++++ b/vendor/github.com/containerd/cri/pkg/config/config.go
+@@ -204,6 +204,9 @@ type PluginConfig struct {
+ 	// DisableProcMount disables Kubernetes ProcMount support. This MUST be set to `true`
+ 	// when using containerd with Kubernetes <=1.11.
+ 	DisableProcMount bool `toml:"disable_proc_mount" json:"disableProcMount"`
++        // ProcessRLimitNoFile sets the maximum file descriptors each container
++        // process can use.
++        ProcessRLimitNoFile int `toml:"process_rlimit_no_file" json:"process_rlimit_no_file"`
+ }
+ 
+ // X509KeyPairStreaming contains the x509 configuration for streaming
+@@ -271,6 +274,7 @@ func DefaultConfig() PluginConfig {
+ 		},
+ 		MaxConcurrentDownloads: 3,
+ 		DisableProcMount:       false,
++                ProcessRLimitNoFile:    1048576,
```

Comment:
Should we match what we're doing for [Docker](https://github.com/bottlerocket-os/bottlerocket/blob/1b8721a72784bfd93f25c95a56f62ee8c7649616/packages/docker-engine/daemon.json#L9) (soft: 1024, hard: 4096) or [containerd's default](https://github.com/containerd/containerd/blob/656b487d333db8ad7134d26cfe12358c46c2b7f8/oci/spec.go#L158-L160) (1024 for both) instead?

Reply:
Those limits seem too low for databases, and there are reports that Elasticsearch doesn't work without more file descriptors.

I would also tend toward a larger number for Docker, and perhaps override this in `host-ctr` for containers launched in that path.

Maybe `65536` soft and `1048576` hard?

Reply:
> I would also tend toward a larger number for Docker

For the ECS variant, the limits can be adjusted in the task definition.

Reply:
Would we want to provide config struct members for soft/hard?
```
...
ProcessRLimitNoFileSoft:    65536,
ProcessRLimitNoFileHard:    1048576,
...
```

Reply:
Added config struct members for soft/hard - but since we're using these default limits and not overriding them anywhere, the node shows `65536` for it's open files limit.  Will this be good enough?